src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs

```diff
@@ -1,28 +1,112 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Buffers.Binary;
 using System.Diagnostics;
 using System.Diagnostics.CodeAnalysis;
 using System.Reflection;
-using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
 using System.Runtime.Serialization;
 using System.Runtime.Versioning;
-using System.Threading;
 
 namespace System.Runtime.CompilerServices
 {
     public static partial class RuntimeHelpers
     {
         [Intrinsic]
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        public static extern void InitializeArray(Array array, RuntimeFieldHandle fldHandle);
+        public static unsafe void InitializeArray(Array array, RuntimeFieldHandle fldHandle)
+        {
+            IRuntimeFieldInfo fldInfo = fldHandle.GetRuntimeFieldInfo();
 
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        private static extern unsafe void* GetSpanDataFrom(
+            if (array is null)
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.array);
+
+            if (!RuntimeFieldHandle.GetRVAFieldInfo(new QCallFieldHandle(ref fldInfo), out IntPtr address, out uint size))
```

Comment:
```suggestion
            if (!RuntimeFieldHandle.GetRVAFieldInfo(fldInfo.Value, out IntPtr address, out uint size))
```
We do not really need `QCallFieldHandle` for this method since the caller must be keeping the field alive to keep the returned field address alive.

Reply:
Is using `QCallHandle` a default pattern that we want to follow and standardize?

Reply:
QCall handles are convenience helpers to avoid writing boilerplate code to deal with unloadability. 

There are situations like this one where the QCall handle is not helping and just adds overhead. I think it is fine to omit QCall handle in situations like this one. We have number of existing QCalls that take direct pointers instead of QCall handles.